roles/kafka_controller/tasks/main.yml

```diff
@@ -0,0 +1,460 @@
+---
+- include_role:
+    name: common
+  when: not common_role_completed|bool
+  tags: common
+
+- name: Gather OS Facts
+  setup:
+    # Only gathers items in list, filters out the rest
+    filter: "{{item}}"
+    gather_subset:
+      - '!all'
+  loop:
+    - ansible_os_family
+    - ansible_fqdn
+    - ansible_distribution
+
+# datadir is not supported way to provide data directories for kafka brokers
+- name: Assert that datadir is not present in the inventory
+  assert:
+    that: >
+      'kafka_controller' not in hostvars[item] or
+      hostvars[item].kafka_controller == None or
+      'datadir' not in hostvars[item].kafka_controller
+    fail_msg: |
+      "Providing 'datadir' in inventory file is no more supported way to configure data directories for kafka brokers."
+      "Please use 'log.dirs' under kafka_controller_custom_properties with comma separated directories."
+  loop: "{{ groups['kafka_controller'] }}"
+
+# To avoid a catastrophic user error where root directory permissions get recursively changed
+- name: Assert log.dirs Property not Misconfigured
+  assert:
+    that:
+      - kafka_controller_final_properties['log.dirs'].split(',')[0] != "/"
```

Comment:
Let's add one more check here, comma separated directories (BYOD) are not supported in Kraft yet.